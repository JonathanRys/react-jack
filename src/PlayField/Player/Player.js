import React, { Component } from "react";
import PropTypes from "prop-types";

import Hand from "../Hand/Hand";

export default class Player extends Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.drawnCard && nextProps.isPlaying) {
      nextProps.takeCard(nextProps.drawnCard);
    }
  }

  render() {
    return this.props.hands.map((hand, index) => (
      <Hand key={`hand-${index}`} hand={hand} dealer={false} />
    ));
  }
}

Player.propTypes = {
  hands: PropTypes.array.isRequired,
  drawnCard: PropTypes.string,
  takeCard: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired
};
