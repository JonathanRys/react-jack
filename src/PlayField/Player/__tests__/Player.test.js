import React from "react";
import { shallow } from "enzyme";

import Player from "../Player";

const mockProps = {
  hands: [["SA", "SJ"]],
  drawnCard: null,
  takeCard: jest.fn(),
  clearCard: jest.fn(),
  isPlaying: true
};

describe("Test Hand component", () => {
  it("matches the snapshot", () => {
    const testProps = { ...mockProps };
    const component = shallow(<Player {...testProps} />);
    expect(component).toMatchSnapshot();
  });

  it("matches the snapshot after drawing a card", () => {
    const testProps = { ...mockProps };
    testProps.drawnCard = "S7";
    const component = shallow(<Player {...testProps} />);
    expect(component).toMatchSnapshot();
  });

  it("doesn't draw a card if not playing", () => {
    const testProps = { ...mockProps };
    testProps.drawnCard = "S7";
    testProps.isPlaying = false;
    const component = shallow(<Player {...testProps} />);
    expect(component).toMatchSnapshot();
  });
});
