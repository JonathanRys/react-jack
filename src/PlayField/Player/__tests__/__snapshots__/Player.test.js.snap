// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hand component doesn't draw a card if not playing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    clearCard={[MockFunction]}
    drawnCard="S7"
    hands={
      Array [
        Array [
          "SA",
          "SJ",
        ],
      ]
    }
    isPlaying={false}
    takeCard={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Hand
      dealer={false}
      hand={
        Array [
          "SA",
          "SJ",
        ]
      }
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Hand
        dealer={false}
        hand={
          Array [
            "SA",
            "SJ",
          ]
        }
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Hand component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    clearCard={[MockFunction]}
    drawnCard={null}
    hands={
      Array [
        Array [
          "SA",
          "SJ",
        ],
      ]
    }
    isPlaying={true}
    takeCard={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Hand
      dealer={false}
      hand={
        Array [
          "SA",
          "SJ",
        ]
      }
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Hand
        dealer={false}
        hand={
          Array [
            "SA",
            "SJ",
          ]
        }
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Test Hand component matches the snapshot after drawing a card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    clearCard={[MockFunction]}
    drawnCard="S7"
    hands={
      Array [
        Array [
          "SA",
          "SJ",
        ],
      ]
    }
    isPlaying={true}
    takeCard={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Hand
      dealer={false}
      hand={
        Array [
          "SA",
          "SJ",
        ]
      }
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Hand
        dealer={false}
        hand={
          Array [
            "SA",
            "SJ",
          ]
        }
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
